### PR TITLE
"A Tale of Two `rand`s" submission for Advent 2017 (Dec 5th)

### PR DESCRIPTION
**Pull request by nelz9999**

### CODE REVIEW

File: content/advent-2017/a-tale-of-two-rands.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+linktitle = "two rands"
+date = 2017-11-10T06:40:42Z
+series = ["Advent 2017"]
+draft = true
++++
+
+I had always been a bit confused as to how the `crypto/rand` package and the `math/rand` package were related, or how they were expected to work (together). Is this something that everyone else already grokked, or is that just my impostor syndrome talking? Well, one day I decided to see if I could defeat my ignorance, and this blog post is the result of that investigation.
+
+# The `math` One
+If you've every poked around in the [`math/rand`](https://golang.org/pkg/math/rand) package, you might agree that it presents a fairly convenient API. My favorite example is the `func Intn(n int) int`, a function that returns a random number within the range that you've given it. SUPER USEFUL!
```
Comment (vdemario) [inline content/advent-2017/a-tale-of-two-rands.md]: Small typo: s/every/ever

File: content/advent-2017/a-tale-of-two-rands.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+You may be asking about the difference between the top-level functions and the functions hung off an instance of the `Rand` type. If you look at the [source code](https://golang.org/src/math/rand/rand.go), you'll see that the top-level functions are just convenience wrappers that refer to a globally instantiated package value called `globalRand`.
+
+There are a few gotchas when using this package, though. The basic usage of only provides __pseudo-random__ numbers, as a function of the seed. This means that if you create two `Rand` instances using a functionally equivalent seed, equivalent calls (in order and function) to the two instances will produce _parallel_ outputs. (I found this concept to be personally challenging to my understanding of "random", because I wouldn't expect to be able to anticipate a "random" result.) If the two `Rand` instances are seeded with different values, the parallel behavior will not be observed.
```
Comment (vdemario) [inline content/advent-2017/a-tale-of-two-rands.md]: I think there's an extra _of_ here, it should be

> The basic usage only provides __pseudo-random__ numbers, as a function of the seed.

File: content/advent-2017/a-tale-of-two-rands.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+# The `crypto` One
+Now, let's look at [`crypto/rand`](https://golang.org/pkg/crypto/rand/). Okay, it's got a nice and concise API surface. The only thing is: HOW THE HECK DO I USE THIS?!? I see that I can generally get byte slices of random 1's and 0's, but what do I do with those?!? That's not nearly as useful as what `math/rand` provides, right?
+
+Hrm. Maybe the question is: How can I combine these two wildly different packages?
```
Comment (vdemario) [inline content/advent-2017/a-tale-of-two-rands.md]: By this point in the text, I wouldn't expect that question. Maybe it gets clearer after reading the rest, but to me it feels like there should be a little bit more explanation about what crypto/rand does and what it's for before it makes sense to combine the two.

**Comment by nelz9999** [comment]
@vdemario Thanks for the feedback! Update coming in a minute!

**Comment by dgryski** [comment]
@vdemario Most of the readers probably don't have English as their first language either, so it's nice to have a review from non-native speakers for comprehension and clarity.

**Comment by vdemario** [comment]
@dgryski Thanks, I'll try to do more reviews on other posts too if I can.

**Pull request merged**